languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -10069,6 +10069,40 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+      <rulegroup id="ACCORD_TOUT_ADJECTIF" name="accord tout adjectif" default="temp_off">
+          <rule>
+              <antipattern>
+                  <token>tout</token>
+                  <token regexp="yes">publique?|public?</token>
+              </antipattern>
+              <pattern>
+                  <token postag="(P\+)?D.*" postag_regexp="yes"/>
+                  <token postag="N f s" postag_regexp="yes"/>
+                  <marker>
+                      <token>tout</token>
+                  </marker>
+                  <token postag="J f s" postag_regexp="yes"><exception regexp="yes">^[aâeéèêioôœuhy].*|&H_ASPIRE;</exception></token>
+              </pattern>
+              <message>Vérifiez l’accord entre « \1 » et « \3 ».</message>
+              <suggestion>toute</suggestion>
+              <example correction="toute">Une veste <marker>tout</marker> neuve.</example>
+              <example>Une veste <marker>tout</marker> publique.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <token postag="(P\+)?D.*" postag_regexp="yes"/>
+                  <token postag="N f p" postag_regexp="yes"/>
+                  <marker>
+                      <token regexp="yes">toute?</token>
+                  </marker>
+                  <token postag="J f p" postag_regexp="yes"><exception regexp="yes">^[aâeéèêioôœuhy].*|&H_ASPIRE;</exception></token>
```

Comment:
Same thing here: it makes sense to include [ef] and s?p, right? Do you see any FPs?

```xml
          <rule>
              <pattern>
                  <token postag="(P\+)?D.*" postag_regexp="yes"/>
                  <token postag="N f p" postag_regexp="yes"/>
                  <marker>
                      <token regexp="yes">toute?</token>
                  </marker>
                  <token postag="J [ef] s?p" postag_regexp="yes"><exception regexp="yes">^[aâeéèêioôœuhy].*|&H_ASPIRE;</exception></token>
              </pattern>
              <message>Vérifiez l’accord entre « \1 » et « \3 ».</message>
              <suggestion>toutes</suggestion>
              <example correction="toutes">Des vestes <marker>tout</marker> neuves.</example>
              <example correction="toutes">Ils s'adressent aux fleurs <marker>toute</marker> colorées.</example>
          </rule>
```

Reply:
I have done all the changes required so I think the rule is ready.

Reply:
I've found this one:

> \- Rocher des abeilles toute petite barre (10-15 voies) mais dans un site sympa

It's not a super-plausible sentence, but it shows that an AP is needed for the case that "tout + adjectif" is followed by a noun that matches in gender and number